pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -612,18 +611,46 @@ func (r *hibernationReconciler) checkClusterRunning(cd *hivev1.ClusterDeployment
 		if err := r.updateClusterDeploymentStatus(cd, logger); err != nil {
 			return reconcile.Result{}, err
 		}
-		poolNS, poolName := "", ""
-		if cd.Spec.ClusterPoolRef != nil {
-			poolNS = cd.Spec.ClusterPoolRef.Namespace
-			poolName = cd.Spec.ClusterPoolRef.PoolName
-		}
 		// logging with time since hibernating condition was set to ResumingOrRunning state
-		hivemetrics.MetricClusterReadyTransitionSeconds.WithLabelValues(cd.Labels[constants.VersionMajorMinorPatchLabel],
-			cd.Labels[hivev1.HiveClusterPlatformLabel], poolNS, poolName).Observe(time.Since(hibernatingCondition.LastTransitionTime.Time).Seconds())
+		logMetric(hivemetrics.MetricClusterReadyTransitionSeconds, cd,
+			time.Since(hibernatingCondition.LastTransitionTime.Time).Seconds())
+		// Clear entry from currently resuming clusters if exists
+		deleteMetric(hivemetrics.MetricResumingClustersSeconds, cd)
 	}
 	return reconcile.Result{}, nil
 }
 
+// deleteMetric has the knowledge of labels needed to clear the metrics for currently transitioning clusters
+// Note: do not use for cumulative metrics - they have different labels and are not designed to be cleared
+func deleteMetric(metric *prometheus.HistogramVec, cd *hivev1.ClusterDeployment) {
+	poolNS := "unknown"
+	if cd.Spec.ClusterPoolRef != nil {
+		poolNS = cd.Spec.ClusterPoolRef.Namespace
+	}
+	metric.Delete(map[string]string{
+		"cluster_deployment_namespace": cd.Namespace,
+		"cluster_deployment":           cd.Name,
+		"platform":                     cd.Labels[hivev1.HiveClusterPlatformLabel],
+		"cluster_version":              cd.Labels[constants.VersionMajorMinorPatchLabel],
+		"cluster_pool_namespace":       poolNS,
+	})
+}
+
+// logMetric should be used to log cumulative metrics
+func logMetric(metric *prometheus.HistogramVec, cd *hivev1.ClusterDeployment, time float64) {
+	poolNS, poolName := "unknown", "unknown"
+	if cd.Spec.ClusterPoolRef != nil {
+		poolNS = cd.Spec.ClusterPoolRef.Namespace
+		poolName = cd.Spec.ClusterPoolRef.PoolName
+	}
+	if hivemetrics.ShouldLogDurationMetric(metric, time) {
```

Comment:
Suggest putting this short-circuit at the top of the function so we're not doing extra work (albeit negligible) if we're not going to log.
```suggestion
	if !hivemetrics.ShouldLogDurationMetric(metric, time) { return }
```